modtree/modtree
# fix: swagger base URL

### Summary of changes

Swagger base URL can now be controlled by env var `NEXT_PUBLIC_SWAGGER`.

### Implementation

The key issue here is that when we serve docs locally, we want it to point to our local backend. But when we serve docs in production, we want it to point to the dev backend (not production, because of security). Here are 2 ideas:

1. Export 2 sets of docs, where only the `baseUrl` is different.

The docs deployments (local/production) simply select the correct file. The environment could theoretically be inferred from the URL of the docs page, but it's probably good to use an env var for this. Hence this method will involve the use of env vars.

However, there is duplication, and the docs deployment would include both files.

2. Export 1 set of docs, but modify `baseUrl` in the Swagger component.

This also requires env vars to find out whether the deployment is local or production. This modifies the docs before displaying it, so only 1 set is needed. Modification may not be a good idea, but I think it is tolerable for a small change.

Both methods involve env vars, but Method 2 does not have duplication. Hence I chose Method 2.

### Testing

- Add `NEXT_PUBLIC_SWAGGER` to docs deploy, pointing to dev backend
- Push to `/d/docs` to test docs deploy
- Go to the Swagger tab in the docs. The server should point to dev backend.
- Test GET `/module/CS1010`. Should get a 200 response
